binding/react/src/use_rhino.ts

```diff
@@ -9,277 +9,132 @@
-  const [isError, setIsError] = useState<boolean | null>(null);
-  const [isListening, setIsListening] = useState(false);
-  const [isLoaded, setIsLoaded] = useState(false);
-  const [isTalking, setIsTalking] = useState(false);
+  error: Error | string | null;
+  init: (
+    accessKey: string,
+    context: RhinoContext,
+    model: RhinoModel,
+    options?: RhinoOptions
+  ) => Promise<void>;
+  process: () => Promise<void>;
+  release: () => Promise<void>;
+  } {
+  const rhinoRef = useRef<RhinoWorker | null>(null);
+  const [inference, setInference] = useState<RhinoInference | null>(null);
   const [contextInfo, setContextInfo] = useState<string | null>(null);
-  const [rhinoWorker, setRhinoWorker] = useState<RhinoWorker | null>(null);
-  const [
-    webVoiceProcessor,
-    setWebVoiceProcessor,
-  ] = useState<WebVoiceProcessor | null>(null);
-
-  const start = (): boolean => {
-    if (webVoiceProcessor !== null) {
-      webVoiceProcessor.start().then(() => {
-        setIsListening(true);
-        return true;
-      });
-    }
-    return false;
-  };
-
-  const pause = (): boolean => {
-    if (webVoiceProcessor !== null) {
-      webVoiceProcessor.pause();
-      setIsListening(false);
-      return true;
-    }
-    return false;
-  };
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [isListening, setIsListening] = useState(false);
+  const [error, setError] = useState<Error | string | null>(null);
 
-  const stop = (): boolean => {
-    if (webVoiceProcessor !== null) {
-      webVoiceProcessor.stop().then(() => {
+  const inferenceCallback = useCallback(
+    (newInference: RhinoInference): void => {
+      if (newInference && newInference.isFinalized) {
+        if (rhinoRef.current) {
+          WebVoiceProcessor.unsubscribe(rhinoRef.current);
+        }
         setIsListening(false);
-        return true;
-      });
-    }
-    return false;
-  };
-
-  const pushToTalk = (): boolean => {
-    if (webVoiceProcessor !== null && rhinoWorker !== null) {
-      setIsTalking(true);
-      rhinoWorker.postMessage({ command: 'resume' });
-      return true;
-    }
-    return false;
-  };
-
-  /** Refresh the inference callback when it changes (avoid stale closure) */
-  useEffect(() => {
-    if (typeof inferenceCallback !== 'function') {
-      // eslint-disable-next-line no-console
-      console.warn('useRhino: inferenceCallback is not a function');
-    }
+        setInference(newInference);
+      }
+    },
+    []
+  );
+
+  const errorCallback = useCallback((newError: any): void => {
+    setError(newError);
+  }, []);
+
+  const init = useCallback(
+    async (
+      accessKey: string,
+      context: RhinoContext,
+      model: RhinoModel,
+      options: RhinoOptions = {}
+    ): Promise<void> => {
+      if (options.processErrorCallback) {
+        // eslint-disable-next-line no-console
+        console.warn(
+          "'processErrorCallback' options is not supported, use the 'error' hook variable instead."
```

Comment:
Kwangsoo and I used: 
console.warn("'processErrorCallback' is only supported in the Rhino Web SDK. Use the 'error' state to monitor for errors in the React SDK.");